src/main/java/net/tcpshield/tcpshield/fabric/impl/FabricConfigImpl.java

```diff
@@ -0,0 +1,110 @@
+        this.debug = config.debugMode;
+    }
+
+    /**
+     * @return  whether the file specified in {@link FabricConfigImpl#configLocation} exists.
+     */
+    private boolean configExists() {
+        return Files.exists(configLocation);
+    }
+
+    /**
+     * Creates the default configuration file at the location specified in {@link FabricConfigImpl#configLocation}.
+     *
+     * <p>
+     * This operation will always override an existing configuration file if it exists.
+     * Callers should check {@link FabricConfigImpl#configExists()} if they wish to avoid this behavior.
+     */
+    private void createDefaultConfig() {
+        // Ensure the parent directory of our config file exists.
+        // If it does not exist, attempt to create it now.
+        Path parentDirectory = configLocation.getParent();
+        if (!Files.exists(parentDirectory) || !Files.isDirectory(parentDirectory)) {
+            parentDirectory.toFile().mkdirs();
+        }
+
+        // Copy the config.yml data from our mod jar to the loader config folder.
+        try (InputStream in = getClass().getClassLoader().getResourceAsStream("config.yml")) {
+            Files.copy(in, configLocation);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private ConfigData loadConfig() {
+        try {
+            List<String> strings = Files.readAllLines(configLocation);
+
+            boolean onlyAllowProxyConnections = true;
+            boolean enableGeyserCompatibility = false;
+            boolean debugMode = false;
+
+            // Rudimentary config parsing
+            for (String line : strings) {
+                String[] keyValue = line.split(": ");
+
+                // A config option will only be valid if it is in the format 'a: b'.
+                // Ensure that is the case now.
+                if (keyValue.length == 2) {
+                    String key = keyValue[0];
+                    String value = keyValue[1];
+
+                    switch (key) {
+                        case "only-allow-proxy-connections":
+                            onlyAllowProxyConnections = Boolean.parseBoolean(value);
+                            break;
+                        case "enable-geyser-compatibility":
+                            enableGeyserCompatibility = Boolean.parseBoolean(value);
+                            break;
+                        case "debug-mode":
+                            debugMode = Boolean.parseBoolean(value);
+                            break;
+                    }
+                }
+            }
+
+            return new ConfigData(onlyAllowProxyConnections, enableGeyserCompatibility, debugMode);
+        } catch (IOException e) {
+            throw new RuntimeException("Couldn't load config", e);
+        }
+    }
+
+    private static class ConfigData {
+
+        protected boolean onlyAllowProxyConnections;
+        protected boolean enableGeyserCompatibility;
```

Comment:
As there's no Floodgate on Fabric, this configuration can be left out entirely

Reply:
I have removed the geyser parsing, but the config file still has the field due to the template `config.yml` file being copied over. Would you prefer I leave it as-is, prevent the geyser field from being written in the Fabric config class, or add an extra config file template without the geyser field?

Reply:
I suppose it's fine this way.